src/blockrelay/graphene.cpp

```diff
@@ -335,7 +338,9 @@ bool CGrapheneBlock::HandleMessage(CDataStream &vRecv, CNode *pfrom, std::string
     int nSizeGrapheneBlock = vRecv.size();
     CInv inv(MSG_BLOCK, uint256());
 
-    CGrapheneBlock grapheneBlock(pfrom->xVersion.as_u64c(XVer::BU_GRAPHENE_VERSION_SUPPORTED));
+    CGrapheneBlock grapheneBlock(
+        (int)std::min(GRAPHENE_MAX_VERSION_SUPPORTED, pfrom->xVersion.as_u64c(XVer::BU_GRAPHENE_VERSION_SUPPORTED)),
+        NegotiateFastFilterSupport(pfrom));
```

Comment:
If NegotiateFastFilterSupport throws the incompatible exception here then the exception handler outside of the network processing loop will catch that so we are fine